### ISSUE TITLE
docs: variable-syntax rule incorrectly claims no overlap with enforce-canonical

### ISSUE THREAD

**Issue by ftzi** [sergioazoc/oxlint-tailwindcss#152]
The `enforce-consistent-variable-syntax` documentation says `enforce-canonical` does not touch variable syntax, but version 1.11.0 reports the same variable-shorthand replacements from both rules.

## Documentation mismatch

The [Interactions with other rules section](https://oxlint-tailwindcss.pages.dev/rules/enforce-consistent-variable-syntax#interactions-with-other-rules) states:

> doesn't touch variable syntax. Canonical normalizes utility names; this rule normalizes the variable form. Both run together cleanly.

This claim also appears in the [English source partial](https://github.com/sergioazoc/oxlint-tailwindcss/blob/main/packages/docs/rules/_extras/enforce-consistent-variable-syntax.md).

## Reproduction

Verified with `oxlint-tailwindcss@1.11.0`, `tailwindcss@4.3.3`, and `oxlint@1.75.0` in an isolated temporary project, using a separate plugin cache.

Install these exact versions:

```sh
npm install --save-dev oxlint@1.75.0 oxlint-tailwindcss@1.11.0 tailwindcss@4.3.3
```

`styles.css`:

```css
@import "tailwindcss";
```

`sample.jsx`:

```jsx
export const sample = <div className="bg-[var(--custom-color)] w-[var(--custom-width)]" />;
```

`.oxlintrc.json` (set `entryPoint` to the absolute path of the file above):

```json
{
  "jsPlugins": ["oxlint-tailwindcss"],
  "categories": { "correctness": "off" },
  "rules": {
    "tailwindcss/enforce-canonical": "error",
    "tailwindcss/enforce-consistent-variable-syntax": "error"
  },
  "settings": {
    "tailwindcss": {
      "entryPoint": "/absolute/path/to/styles.css"
    }
  }
}
```

```sh
npx oxlint --disable-nested-config --config .oxlintrc.json sample.jsx
```

## Actual result

Four diagnostics: each rule reports both classes and recommends the same replacements.

| Input | `enforce-canonical` | `enforce-consistent-variable-syntax` |
| --- | --- | --- |
| `bg-[var(--custom-color)]` | `bg-(--custom-color)` | `bg-(--custom-color)` |
| `w-[var(--custom-width)]` | `w-(--custom-width)` | `w-(--custom-width)` |

With only `enforce-canonical` enabled, the same two replacements are still reported. No theme-token or unnecessary-arbitrary-value rule is enabled in either run.

The [canonical rule implementation](https://github.com/sergioazoc/oxlint-tailwindcss/blob/main/packages/oxlint-tailwindcss/src/rules/enforce-canonical.ts) routes dynamic utility values through `canonicalizeClassesSync` and accepts CSS-equivalent results; it does not exclude variable-syntax conversions.

## Expected documentation

Document this overlap and explain when the dedicated variable-syntax rule adds coverage or a different policy. If the intended contract is disjoint diagnostics, clarify that contract and track the implementation mismatch instead. These examples establish overlap, not that the rules have identical coverage.

Rule pages are generated, so a documentation correction belongs in the EN/ES `_extras` sources followed by regeneration.

## AI disclosure

The user requested this issue. GPT-6 Astra investigated the documentation mismatch, ran the isolated CLI reproduction, and wrote this issue.

**Comment by sergioazoc** [comment]
Thanks for this, @ftzi, and thanks for the level of detail. The isolated repro with pinned versions made it trivial to confirm.

You were right, and it turned out to be more than a docs mismatch. `enforce-canonical` really does rewrite the variable shorthand (`bg-[var(--custom-color)]` → `bg-(--custom-color)`), because that's the canonical form Tailwind's `canonicalizeCandidates` returns. So with both rules on you got the duplicate diagnostic you saw.

There was a worse case hiding behind it, too. With `enforce-consistent-variable-syntax` set to `syntax: 'explicit'`, the two autofixes fight each other: canonical pushes to shorthand, the dedicated rule pushes back to explicit. An oscillation.

So instead of just fixing the docs, I fixed the behavior. `enforce-canonical` now cedes the plain `[var(--x)]` ↔ `(--x)` swap to `enforce-consistent-variable-syntax` (the single owner of variable-syntax policy), the same way it already cedes v3 renames to `no-deprecated-classes`. It still handles the var cases that aren't a plain swap: value→token like `rounded-[var(--radius-sm)]` → `rounded-sm`, and opacity-modifier forms like `text-[var(--color-text)]/90`. No more double diagnostic, no more `explicit`-mode conflict.

Your report also sent me auditing the rest of the docs, and I found a few other claims that had drifted from the code, so those got fixed too.

Shipped in v1.12.0. Thanks again for the sharp report.
